internal/shell/shell.go

```diff
@@ -200,6 +196,9 @@ func (sh *Shell) appendStatementPartAndExecuteIfFinished(statementPart string) {
 		sh.state.readline.SetPrompt(sh.promptFmt(promptNewStatement))
 		err := sh.db.ExecuteAndPrintStatements(completeStatement, sh.config.OutF, false, sh.state.printMode)
 		if err != nil {
+			if strings.Contains(err.Error(), "context canceled") {
+				err = &shellerrors.CancelQueryContextError{}
+			}
```

Comment:
We already have a place for checks like this. Right now we check for `"interactive transaction not allowed in HTTP queries"` inside `db.executeQuery`. Is there any reason to do this check here instead of putting it together with the one that we already have?
Putting it here will make `context canceled` be displayed when we use `--exec` flag

Reply:
The `context canceled` error was not being caught there. So I added it here, but since we also need it when we use `--exec` flag, I move it to the return of `PrintStatementsResult`.